src/main/java/org/kohsuke/github/extras/auth/JWTTokenProvider.java

```diff
@@ -0,0 +1,132 @@
+package org.kohsuke.github.extras.auth;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.kohsuke.github.AuthorizationProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.Date;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A authorization provider that gives valid JWT tokens. These tokens are then used to create a time-based token to
+ * authenticate as an application. This token provider does not provide any kind of caching, and will always request a
+ * new token to the API.
+ */
+public class JWTTokenProvider implements AuthorizationProvider {
+
+    private static final long MINUTES_10 = Duration.ofMinutes(10).toMillis();
+
+    private final PrivateKey privateKey;
+
+    @Nonnull
+    private Instant validUntil = Instant.MIN;
+
+    private String token;
+
+    /**
+     * The identifier for the application
+     */
+    private final String applicationId;
+
+    public JWTTokenProvider(String applicationId, File keyFile) throws GeneralSecurityException, IOException {
+        this(applicationId, loadPrivateKey(keyFile.toPath()));
+    }
+
+    public JWTTokenProvider(String applicationId, Path keyPath) throws GeneralSecurityException, IOException {
+        this(applicationId, loadPrivateKey(keyPath));
+    }
+
+    public JWTTokenProvider(String applicationId, PrivateKey privateKey) {
+        this.privateKey = privateKey;
+        this.applicationId = applicationId;
+    }
+
+    @Override
+    public String getEncodedAuthorization() throws IOException {
+        synchronized (this) {
+            if (Instant.now().isAfter(validUntil)) {
+                token = refreshJWT();
+            }
+            return String.format("Bearer %s", token);
```

Comment:
@MarcosCela 
Could you add a test that verifies this?

Reply:
Will do!